pip/download.py

```diff
@@ -198,6 +203,31 @@ def parse_credentials(self, netloc):
             return userinfo, None
         return None, None
 
+    @classmethod
```

Comment:
Since `authlib` is only called on an instance AFAICT, I'm not sure why it needs to be a class method?